webui/react/src/pages/Authentication.tsx

```diff
@@ -0,0 +1,156 @@
+import { Button, Form, Input } from 'antd';
+import { message } from 'antd';
+import axios from 'axios';
+import queryString from 'query-string';
+import React, { useEffect, useState } from 'react';
+import { useHistory, useLocation } from 'react-router';
+import { Redirect } from 'react-router-dom';
+
+import Icon from 'components/Icon';
+import Logo, { LogoTypes } from 'components/Logo';
+import Spinner from 'components/Spinner';
+import Auth, { updateAuth } from 'contexts/Auth';
+import handleError, { ErrorType } from 'ErrorHandler';
+import { crossoverRoute, isCrossoverRoute } from 'routes';
+import { isLoginFailure, login, logout } from 'services/api';
+import { Credentials } from 'types';
+
+import css from './Authentication.module.scss';
+
+const DEFAULT_REDIRECT = '/det/dashboard';
+
+type WithSearch<T> = T & {location: {search: string}};
+interface Queries {
+  redirect?: string;
+}
+
+interface FromValues {
+  password?: string;
+  username?: string;
+}
+
+const Authentication: React.FC<WithSearch<{}>> = (props: WithSearch<{}>) => {
+  const history = useHistory();
+  const location = useLocation();
+  const auth = Auth.useStateContext();
+  const setAuth = Auth.useActionContext();
+  const [ isLoading, setIsLoading ] = useState(true);
+  const [ badCredentials, setBadCredentials ] = useState(false);
+  const [ canSubmit, setCanSubmit ] = useState(false);
+
+  const queries: Queries = queryString.parse(props.location.search);
+
+  const redirectUri= queries.redirect || DEFAULT_REDIRECT;
+
+  useEffect(() => {
+    const source = axios.CancelToken.source();
+    updateAuth(setAuth, source.token).then(() => setIsLoading(false));
+    return (): void => {
+      source.cancel();
+    };
+  }, [ setAuth ]);
+
+  const isLogout = location.pathname.endsWith('logout');
+  if (isLogout) {
+    logout({});
+    setAuth({ type: Auth.ActionType.Reset });
+    history.push('/det/login' + props.location.search);
+  }
+
+  const onFinish = (creds: FromValues): void => {
+    // TODO validate the creds type?
+    setCanSubmit(false);
+    const hideLoader = message.loading('logging in..');
+    login(creds as Credentials)
+      .then(() => updateAuth(setAuth))
+      .catch((e: Error) => {
+        setBadCredentials(isLoginFailure(e));
+        if (badCredentials) return;
+        // DISCUSS do we want to potentially report it?
+        // or pass it through error handler for some reason?
```

Comment:
this could be tied to the telemetry ticket instead, but after discussion with product, it will be valuable to capture data around if people are struggling with webui login.

Reply:
for now, I enable passing it through error handler with the silent flag set.